src/renderer/components/deposit/withdraw/AsymWithdraw.tsx

```diff
@@ -0,0 +1,368 @@
+import React, { useState, useMemo, useCallback } from 'react'
+
+import * as RD from '@devexperts/remote-data-ts'
+import { getWithdrawMemo } from '@thorchain/asgardex-util'
+import { Asset, baseAmount, BaseAmount, baseToAsset, Chain, formatAssetAmountCurrency } from '@xchainjs/xchain-util'
+import { Col } from 'antd'
+import BigNumber from 'bignumber.js'
+import * as FP from 'fp-ts/function'
+import * as O from 'fp-ts/lib/Option'
+import { useObservableState } from 'observable-hooks'
+import { useIntl } from 'react-intl'
+import * as Rx from 'rxjs'
+import * as RxOp from 'rxjs/operators'
+
+import { Network } from '../../../../shared/api/types'
+import { ZERO_BASE_AMOUNT } from '../../../const'
+import { eqAsset } from '../../../helpers/fp/eq'
+import { sequenceTOption } from '../../../helpers/fpHelpers'
+import { useSubscriptionState } from '../../../hooks/useSubscriptionState'
+import { INITIAL_WITHDRAW_STATE } from '../../../services/chain/const'
+import { FeeLD, FeeRD, Memo, WithdrawState, WithdrawStateHandler } from '../../../services/chain/types'
+import { PoolAddress } from '../../../services/midgard/types'
+import { ValidatePasswordHandler } from '../../../services/wallet/types'
+import { PasswordModal } from '../../modal/password'
+import { TxModal } from '../../modal/tx'
+import { DepositAssets } from '../../modal/tx/extra'
+import { Fees, UIFeesRD } from '../../uielements/fees'
+import { Label } from '../../uielements/label'
+import { getAsymWithdrawAmount } from './Withdraw.helper'
+import * as Styled from './Withdraw.styles'
+
+export type Props = {
+  /** Asset to withdraw */
+  asset: Asset
+  /** Rune price */
+  runePrice: BigNumber
+  /** Asset price */
+  assetPrice: BigNumber
```

Comment:
could you pls add to the comment that this are prices in BaseAmount value

Reply:
Good point. Sure, I can add a comment, but then we should do this in all places then. Just search for `xyzPrice` and you will find a lot of `xyzPrices` ... The question is: Why do we provide prices in `BigNumber` and not in `BaseAmount` everywhere ;)

Reply:
Issue filed: **Use `BaseAmount` (not BigNumber) for prices** #987

Reply:
yep. good question :)